### PR TITLE
perf(spec-parser): add warning and ignore api only contains optional param

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

**Review by yuqizhou77 — DISMISSED** [review]

File: packages/fx-core/src/common/spec-parser/constants.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ export class ConstantString {
     "Server url is not correct: relative server url is not supported.";
   static readonly ResolveServerUrlFailed =
     "Resolve server Url failed: environment variable %s is not defined";
+  static readonly OperationOnlyContainsOptionalParam =
+    "Operation %s %s only contains optional parameters which is not supported";
```
Comment (yuqizhou77) [inline packages/fx-core/src/common/spec-parser/constants.ts:34]: better to add " or ' for %s

File: packages/fx-core/src/common/spec-parser/manifestUpdater.ts [hunk 2]
```diff
@@ -124,6 +134,19 @@ export async function generateCommands(
...
 
+              if (parameters.length === 0) {
+                warnings.push({
+                  type: WarningType.OperationOnlyContainsOptionalParam,
+                  content: format(
+                    ConstantString.OperationOnlyContainsOptionalParam,
+                    method.toUpperCase(),
+                    pathUrl
+                  ),
+                  data: method.toUpperCase() + " " + pathUrl,
```
Comment (yuqizhou77) [inline packages/fx-core/src/common/spec-parser/manifestUpdater.ts:145]: Can we have operationId here instead?

File: packages/fx-core/src/common/spec-parser/constants.ts (hunk 1 above)
Comment (SLdragon) [reply packages/fx-core/src/common/spec-parser/constants.ts:34]: OK

File: packages/fx-core/src/common/spec-parser/manifestUpdater.ts (hunk 2 above)
Comment (SLdragon) [reply packages/fx-core/src/common/spec-parser/manifestUpdater.ts:145]: sure

Comment (yuqizhou77) [reply packages/fx-core/src/common/spec-parser/manifestUpdater.ts:145]: or even both operationId and "Method + Path"

Comment (SLdragon) [reply packages/fx-core/src/common/spec-parser/manifestUpdater.ts:145]: As discussed, will use operationId

**Review by adashen — DISMISSED** [review]

**Review by yuqizhou77 — APPROVED** [review]

**Review by adashen — APPROVED** [review]

**Pull request merged**